mozilla-release/browser/components/nsBrowserGlue.js

```diff
@@ -2650,6 +2650,55 @@ BrowserGlue.prototype = {
       }
     }
 
+    // CLIQZ-SPECIAL: moderating default browser message popup timings as per DB-2076
+    let defaultBrowserCheckFlag = false;
+    let firstDefaultBrowserCheckTimestamp =
+      Services.prefs.getPrefType("browser.shell.firstDefaultBrowserCheckTimestamp") !== 0 ?
+        Services.prefs.getIntPref("browser.shell.firstDefaultBrowserCheckTimestamp") :
+        null;
+
+    if (!firstDefaultBrowserCheckTimestamp) {
+      firstDefaultBrowserCheckTimestamp = parseInt(Date.now() / 1000);
+      Services.prefs.setIntPref("browser.shell.firstDefaultBrowserCheckTimestamp", firstDefaultBrowserCheckTimestamp);
+    }
+
+    let checkLevel =
+      Services.prefs.getPrefType("browser.shell.defaultBrowserCheckLevel") !== 0 ?
+        Services.prefs.getIntPref("browser.shell.defaultBrowserCheckLevel"):
+        0;
+
+    function setLevel(val) {
+      Services.prefs.setIntPref("browser.shell.defaultBrowserCheckLevel", val);
+      return true;
+    }
+
+    if (firstDefaultBrowserCheckTimestamp && willPrompt) {
+      let whatsNow = parseInt(Date.now() / 1000);
+      let timeDiff = whatsNow - firstDefaultBrowserCheckTimestamp;
```

Comment:
@alver-cliqz 
`if (timeDiff < weekAge) {` this will take care of negative values, even if user changes time (timezones are fine as it will impact only several hours) he/she will see prompt once, beyond that again the cycle will continue.